### PR TITLE
[CCR] UI updates

### PR DESCRIPTION
**Pull request by loixlab**
Added a few UI updates for the CCR app:

- Auto-follow patterns table: we can search now on prefix and suffix
- The indices preview when creating an auto-follow pattern has been changed to a time series
- The indices previews displays in bold the prefix and the suffix

### CODE REVIEW

File: x-pack/package.json [hunk 1]
```diff
@@ -160,6 +160,7 @@
     "d3": "3.5.6",
     "d3-scale": "1.0.6",
     "dataloader": "^1.4.0",
+    "date-fns": "^1.30.1",
```
Comment (cjcenizal) [inline x-pack/package.json:163]: Why was this added?

File: x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern.js [hunk 2]
```diff
@@ -4,8 +4,15 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import React, { Fragment } from 'react';
+import { format, addDays } from 'date-fns';
+
 export const getFollowPattern = (prefix = '', suffix = '', template = '{{leader_index}}') => (
-  prefix + template + suffix
+  <Fragment>
```
Comment (cjcenizal) [inline x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern.js:11]: I have a few suggestions!

1. Could this return an object of the shape `{ prefix, suffix, template }` and can we delegate the rendering logic to the `AutoFollowPatternPreview` component? This way we can keep all rendering logic in the `sections` dir.
2. Can we use `<strong>` instead of `<span>`?
3. I'm on the fence, but do you think it might make more sense to bold the index pattern instead, since it is the part that's changing?

image

File: x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/auto_follow_pattern_table/auto_follow_pattern_table.js [hunk 3]
```diff
@@ -89,29 +90,21 @@ export class AutoFollowPatternTableUI extends PureComponent {
       ),
       render: (leaderPatterns) => leaderPatterns.join(', '),
     }, {
-      field: 'followIndexPattern',
+      field: 'followIndexPatternPrefix',
```
Comment (cjcenizal) [inline x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/auto_follow_pattern_table/auto_follow_pattern_table.js:93]: Can we add `sortable: true` to the prefix and suffix columns?

**Review by cjcenizal — APPROVED** [review]
Looks good! Had a few ideas.

File: x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/auto_follow_pattern_table/auto_follow_pattern_table.js (hunk 3 above)
Comment (loixlab) [reply x-pack/plugins/cross_cluster_replication/public/app/sections/home/auto_follow_pattern_list/components/auto_follow_pattern_table/auto_follow_pattern_table.js:93]: Yes, don't know why I thought it was already set to `true` 😊

File: x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern.js (hunk 2 above)
Comment (loixlab) [reply x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern.js:11]: Ok, good points. About where the bold is, I asked Yaron and he agrees with you.

File: x-pack/package.json (hunk 1 above)
Comment (loixlab) [reply x-pack/package.json:163]: Mainly because

- it's functional
- it's immutable
- it's faster
- we can bundle what we need and not the whole lib 

https://medium.com/@k2u4yt/momentjs-vs-date-fns-6bddc7bfa21e

Comment (cjcenizal) [reply x-pack/package.json:163]: Thanks, I read the link and have a good idea of why date-fns is better than moment. I'd be in support of exploring migrating over. :)

In this particular situation, is it worth adding the extra dependency though? Do we gain anything significant, e.g. noticeable performance improvement? If not, then I think we should just stick with moment (warts and all) to avoid adding another dependency.
